src/include/sof/lib/mailbox.h

```diff
@@ -44,11 +44,11 @@
 static inline
 void mailbox_dspbox_write(size_t offset, const void *src, size_t bytes)
 {
-	int ret = memcpy_s((void *)(MAILBOX_DSPBOX_BASE + offset),
+	int ret = memcpy_s((void *)((uintptr_t)MAILBOX_DSPBOX_BASE + offset),
```

Comment:
The double cast and nesting are a bit of a eyesore... much simpler:
```suggestion
	int ret = memcpy_s((char *)MAILBOX_DSPBOX_BASE + offset),
```

Reply:
@marc-hb why do you think this would help? The difference between your version and the current one is replacing `(void *)` cast with a `(char *)` cast. How is that better? The problem that this commit is fixing is, that `MAILBOX_DSPBOX_BASE` and other similar macros are now derived from Device Tree by Zephyr:
```
#define L2_SRAM_BASE \
	(DT_REG_ADDR(DT_NODELABEL(sram0)))
```
which is apparently 64-bit. And a pointer - either `void *` or `char *` or any other - on Xtensa is 32-bit. So the compiler complains about converting a 64-bit integer value to a 32-bit wide pointer. That's why we need to first convert a 64-bit integer value to a 32-bit integer value, and then convert the latter to a pointer.
PS: `char *` has really nothing to do here and in similar places. This code isn't dealing with characters / strings. `void *` is perfectly fine here or, if, say, you needed to count bytes, you'd use `uint8_t *`, but not a `char *`.

Reply:
> warning: cast to pointer from integer of different size

Sorry, I realize now my suggestion does not solve that problem.

>  you'd use uint8_t *, but not a char *

There's no technical difference between `uint8_t *` and `char *` but you're right again, the former is more readable. Are there still environments that don't define `uint8_t`? Hopefully not.

> This code isn't dealing with characters / strings.

8 bits are too few for a codepoint, so `char` is not even suited to characters and strings either :-D